flink-clients/src/main/java/org/apache/flink/client/deployment/application/ApplicationDispatcherBootstrap.java

```diff
@@ -254,37 +265,67 @@ private void runApplicationEntryPoint(
+                    && maybeDuplicate.get().isGloballyTerminated()) {
+                final JobID jobId = maybeDuplicate.get().getJobID();
+                tolerateMissingResult.add(jobId);
+                jobIdsFuture.complete(Collections.singletonList(jobId));
+            } else {
+                jobIdsFuture.completeExceptionally(
+                        new ApplicationExecutionException("Could not execute application.", t));
+            }
         }
     }
 
     private CompletableFuture<Void> getApplicationResult(
             final DispatcherGateway dispatcherGateway,
             final Collection<JobID> applicationJobIds,
+            final Set<JobID> tolerateMissingResult,
             final ScheduledExecutor executor) {
         final List<CompletableFuture<?>> jobResultFutures =
                 applicationJobIds.stream()
                         .map(
                                 jobId ->
                                         unwrapJobResultException(
-                                                getJobResult(dispatcherGateway, jobId, executor)))
+                                                getJobResult(
+                                                        dispatcherGateway,
+                                                        jobId,
+                                                        executor,
+                                                        tolerateMissingResult.contains(jobId))))
                         .collect(Collectors.toList());
         return FutureUtils.waitForAll(jobResultFutures);
     }
 
     private CompletableFuture<JobResult> getJobResult(
             final DispatcherGateway dispatcherGateway,
             final JobID jobId,
-            final ScheduledExecutor scheduledExecutor) {
-
+            final ScheduledExecutor scheduledExecutor,
+            final boolean tolerateMissingResult) {
         final Time timeout =
                 Time.milliseconds(configuration.get(ClientOptions.CLIENT_TIMEOUT).toMillis());
         final Time retryPeriod =
                 Time.milliseconds(configuration.get(ClientOptions.CLIENT_RETRY_PERIOD).toMillis());
-
-        return JobStatusPollingUtils.getJobResult(
-                dispatcherGateway, jobId, scheduledExecutor, timeout, retryPeriod);
+        final CompletableFuture<JobResult> jobResultFuture =
+                JobStatusPollingUtils.getJobResult(
+                        dispatcherGateway, jobId, scheduledExecutor, timeout, retryPeriod);
+        if (tolerateMissingResult) {
+            // Return "unknown" job result if dispatcher no longer knows the actual result.
+            return FutureUtils.handleException(
+                    jobResultFuture,
+                    FlinkJobNotFoundException.class,
+                    exception ->
+                            new JobResult.Builder()
+                                    .jobId(jobId)
+                                    .applicationStatus(ApplicationStatus.UNKNOWN)
```

Comment:
I think an `ApplicationStatus.UNKNOWN` will result in a `JobExecutionException` when calling `JobResult.toJobExecutionResult`. I think this will fail the client when calling `StreamingExecutionEnvironment.execute()`.

Reply:
Also, we might wanna prevent that the user can call `JobExecutionResult.getAccumulatorResult`. I would suggest that we fail if such a call happens.

Reply:
Hmm, this is tricky. Right now I think we're just failing call to `execute()` with DuplicateJobSubmission, so there should be no more calls from client code. We're basically just ignoring that exception in ApplicationDispatcherBoostrap so cluster can properly cleanup.

Letting user to intetract with the result would probably require some changes to EmbeddedJobClient 🤔 I need to give a closer look, if there is any way we can move the duplication handling there.

Reply:
Sum up from an offline discussion with Till. We can live this limitation for now, until we solve [FLINK-11813](https://issues.apache.org/jira/browse/FLINK-11813). We'll document the limitation in the release notes.